shell/platform/linux/public/flutter_linux/fl_texture_registrar.h

```diff
@@ -0,0 +1,82 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_TEXTURE_REGISTRAR_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_TEXTURE_REGISTRAR_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <glib-object.h>
+#include <stdint.h>
+
+G_BEGIN_DECLS
+
+/**
+ * The pixel buffer copy callback provided to the Flutter engine to copy the
+ * texture.
+ * @width: inout parameter representing the given width and the actual width.
+ * @height: inout parameter representing the given height and the actual height.
+ * @buffer: out parameter representing the pixel buffer.
+ *
+ * Returns: %TRUE on success.
+ */
+typedef gboolean (*FlTextureCallback)(size_t* width,
+                                      size_t* height,
+                                      const uint8_t** buffer,
```

Comment:
This might be better redefined as `void* out`, so it is agnostic of the underlying output and can therefore be safely passed to all `FlExternalTexture` subclasses.